[SPARK-48549][SQL][DOCS][FOLLOWUP] Add migration guide for SQL function `sentences` behavior changes (#48099)
apache/spark: docs/sql-migration-guide.md

- Since Spark 4.0, By default views tolerate column type changes in the query and compensate with casts. To restore the previous behavior, allowing up-casts only, set `spark.sql.legacy.viewSchemaCompensation` to `false`.
- Since Spark 4.0, Views allow control over how they react to underlying query changes. By default views tolerate column type changes in the query and compensate with casts. To disable this feature set `spark.sql.legacy.viewSchemaBindingMode` to `false`. This also removes the clause from `DESCRIBE EXTENDED` and `SHOW CREATE TABLE`.
- Since Spark 4.0, The Storage-Partitioned Join feature flag `spark.sql.sources.v2.bucketing.pushPartValues.enabled` is set to `true`. To restore the previous behavior, set `spark.sql.sources.v2.bucketing.pushPartValues.enabled` to `false`.

> - Since Spark 4.0, the `sentences` function, when `language` parameter is not null and `country` parameter is null, we no longer use `Locale.US`, instead use `Locale(language)`.

Reviewer: Let's follow the convention in the file and use `NULL` instead of just null.

Reviewer: And could you reformulate the sentence and write directly:
```suggestion
- Since Spark 4.0, the `sentences` function uses `Locale(language)` instead of `Locale.US` when `language` parameter is not `NULL` and `country` parameter is `NULL`.
```

Author: Thanks!
